modules/vorbis/audio_stream_ogg_vorbis.cpp

```diff
@@ -110,7 +110,7 @@ int AudioStreamPlaybackOGGVorbis::_mix_frames_vorbis(AudioFrame *p_buffer, int p
 	if (info.channels > 1) {
 		for (int frame = 0; frame < frames; frame++) {
 			p_buffer[frame].l = pcm[0][frame];
-			p_buffer[frame].r = pcm[0][frame];
+			p_buffer[frame].r = pcm[1][frame];
```

Comment:
From what I see in the libvorbis documentation, this might cause an out-of-bounds read for mono files.

https://xiph.org/vorbis/doc/libvorbis/vorbis_synthesis_pcmout.html

Perhaps we could check the number of channels in the vorbis_info? If it's a mono we should keep the current behavior I think. https://xiph.org/vorbis/doc/libvorbis/vorbis_info.html

Reply:
That is exactly what is happening. The info in the if statement is the `vorbis_info` with the already parsed number of channels. The following else condition  is keeping the current behaviour of using the `0th` element for both L and R when we have `<= 1 ` channels

Reply:
Oh gosh you're right, that's what I get for reviewing code before I was fully awake.